src/mpi/init/initthread.c

```diff
@@ -324,11 +324,25 @@ int MPIR_Init_thread(int * argc, char ***argv, int required, int * provided)
     int exit_init_cs_on_failure = 0;
     MPIR_Info *info_ptr;
 
+#if (MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_ARGOBOTS)
+    int rc = ABT_init(*argc, *argv);
+    if (rc != ABT_SUCCESS) {
+        mpi_errno = MPIR_Err_create_code(MPI_SUCCESS, MPIR_ERR_RECOVERABLE,
+                                         "MPI_Init_thread", __LINE__, MPI_ERR_OTHER,
+                                         "**init_argobots_failed", 0);
+        goto fn_fail;
+    }
+    MPL_thread_info_init(0); /* not using ult_yield_only */
+#endif
     /* For any code in the device that wants to check for runtime 
        decisions on the value of isThreaded, set a provisional
        value here. We could let the MPID_Init routine override this */
 #if defined MPICH_IS_THREADED
+#if (MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_ARGOBOTS)
```

Comment:
This check is not needed. Even if it runs with one execution stream, an Argobots application that calls MPI from more than one ULT should ask for MPI_THREADED_MULTIPLE.